### DISCUSSION TITLE
[4.0] no web service media upload yet?

### DISCUSSION THREAD

**Discussion by pjdevries** [joomla/joomla-cms#34224] [General]
Am I missing something, or is there no possibility to upload media files using the web services yet? It would also mean it's not possible to upload article images for instance, which cripples the services quite a bit.

**Comment by alikon** [comment]
> is there no possibility to upload media files using the web services yet? I

correct

**Reply by dgrammatiko** [reply]
@alikon check https://github.com/joomla/joomla-cms/pull/28928 There are some requirements:
- an adapter 
- and the file to be uploaded needs to be base64 encoded

I'm guessing in the model (?) you can handle both and then reach the com_media API

**Reply by pjdevries** [reply]
Like I mentioned below, I'm currently attempting a rough implementation. But I emphasize: no promises. I don't know if I'm skilled enough and if I can spend enough time on it.

My approach is a to have a web service specific model `Joomla\Component\Media\Api\Model\MediaModel` acting as a proxy for com_media's `Joomla\Component\Media\Administrator\Model\ApiModel`. That way I can reuse all or most of com_media's existing functionality.

As soon as I have something that appears to work, I'll create a PR and wait for inevitable criticism :)

**Comment by pjdevries** [comment]
@alikon Thanx for the very to the point answer :D Are there already any thoughts about it though?

**Reply by alikon** [reply]
well imho this feature should be present in 4.0....
unfortunately i've some bad understanding on com_media 🤕

**Reply by pjdevries** [reply]
I agree that it should.
I have no understanding of com_media either, but I'm currently attempting an implementation. No promises though.

**Comment by pjdevries** [comment]
@alikon, @dgrammatiko Here is my attempt: #34314
